### DISCUSSION TITLE
Alternatives to @KafkaListener

### DISCUSSION THREAD

**Discussion by alexp1917** [spring-projects/spring-kafka#1999] [Q&A]
I am trying to make a re-usable kafka consumer library that other teams can use to listen to my messages, complete with dead-letter queue, retry, de/serialization logic, etc... do I need to make a single listener method using `@KafkaListener` and go through hoops associated with that? or is there some information on how i can use the other ways?

Thanks,

**Comment by garyrussell** [comment]
Just implement one of the sub-interfaces of `GenericMessageListener` and create the container programmatically.

https://docs.spring.io/spring-kafka/docs/current/reference/html/#message-listeners
